test/controllers/pair_retro_controller_test.exs

```diff
@@ -15,26 +14,24 @@ defmodule Pairmotron.PairRetroControllerTest do
 
   defp create_user_and_pair_and_retro() do
     user = insert(:user)
-    group = insert(:group, %{owner: user})
-    pair = Pairmotron.TestHelper.create_pair([user], group)
-    retro = create_retro(user, pair)
+    group = insert(:group, %{owner: user, users: [user]})
+    pair = insert(:pair, %{group: group, users: [user]})
+    retro = insert(:retro, user: user, pair: pair)
     {user, pair, retro}
   end
 
-  describe "while authenticated" do
+  describe "using :index while authenticated" do
     setup do
-      user = insert(:user)
-      group = insert(:group, %{owner: user})
-      conn = build_conn() |> log_in(user)
-      {:ok, [conn: conn, logged_in_user: user, group: group]}
+      login_user()
```

Comment:
I ended up extracting all group insertions from setup blocks. Not all tests actually needed a group, so it was an extra db insert in many cases.